redis/lock.py

```diff
@@ -168,7 +178,16 @@ def __exit__(
         exc_value: Optional[BaseException],
         traceback: Optional[TracebackType],
     ) -> None:
-        self.release()
+        try:
+            self.release()
+        except LockNotOwnedError:
+            if self.raise_on_release_error:
+                raise
+            logger.warning("Lock was no longer owned when exiting context manager.")
+        except LockError:
+            if self.raise_on_release_error:
+                raise
+            logger.warning("Lock was unlocked when exiting context manager.")
```

Comment:
Here the suggestion I mentioned in my last comment (please see). Feel free to commit or discard it as you prefer

```suggestion
        except LockError:
            if self.raise_on_release_error:
                raise
            logger.warning("Lock was unlocked or no longer owned when exiting context manager.")
```